presto-main/src/main/java/io/prestosql/execution/scheduler/FileBasedNetworkTopology.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution.scheduler;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airlift.units.Duration;
+import io.prestosql.spi.HostAddress;
+import io.prestosql.spi.PrestoException;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Suppliers.memoize;
+import static com.google.common.base.Suppliers.memoizeWithExpiration;
+import static io.prestosql.spi.StandardErrorCode.CONFIGURATION_INVALID;
+import static java.util.Objects.requireNonNull;
+
+public class FileBasedNetworkTopology
+        implements NetworkTopology
+{
+    private final File networkTopologyFile;
+    private Supplier<Topology> topology;
+
+    @Inject
+    public FileBasedNetworkTopology(NodeSchedulerConfig nodeSchedulerConfig)
+    {
+        networkTopologyFile = new File(requireNonNull(nodeSchedulerConfig.getNetworkTopologyFile()));
+        Duration refreshPeriod = nodeSchedulerConfig.getRefreshPeriod();
+        if (refreshPeriod == null) {
```

Comment:
To simplify the setup and management for admins, the topology should be loaded during construction and validated.  If the file is missing or invalid, throw an exception which will prevent the server from starting.  Then during reload, if the file is missing or invalid, log an error, and keep using the existing value.  This will prevent catching a file in the middle of editing from breaking a running server.  The error should only be logged when going from a valid file to an invalid one, and specifically should not be logged on each refresh (until an valid file is seen again).